fix: complete Tasks GA promotion (docs, site)

mafredri opened the pull request:
## Summary

Completes the Coder Tasks GA promotion by updating swagger tags and regenerating API documentation and updating the frontend API structure.

## Related

Follows #20923 and #20921 which promoted Tasks from Beta/Experimental to GA.

---

🤖 This change was written by Claude Sonnet 4.5 Thinking using [mux](https://github.com/coder/mux) and reviewed by a human 🏂

mafredri committed:
docs: complete Tasks GA promotion

Update swagger tags and refactor frontend to use dedicated TasksApiMethods class.

- Change @Tags Experimental to @Tags Tasks in coderd/aitasks.go
- Extract TasksApiMethods from ExperimentalApiMethods
- Update all API.experimental.* task calls to API.tasks.*

---

🤖 This change was written by Claude Sonnet 4.5 Thinking using [mux](https://github.com/coder/mux) and reviewed by a human 🏂

DanielleMaywood commented on site/src/api/api.ts:
```diff
@@ -440,9 +440,11 @@ export type GetProvisionerDaemonsParams = {
 
 	constructor(protected readonly axios: AxiosInstance) {
 		this.experimental = new ExperimentalApiMethods(this.axios);
+		this.tasks = new TasksApiMethods(this.axios);
```
I think Claude got the wrong impression. The tasks endpoints should be directly inside `ApiMethods`, not under a separate `TasksApiMethods`

mafredri committed:
refactor: move task methods directly into ApiMethods

As per review feedback, task methods should be directly in ApiMethods
rather than a separate TasksApiMethods class.
